File: EmployeeWatch/Domain/Task.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Domain;

[Table("tasks")]
public sealed class Task : Entity<int>
{
    [Required]
    [Key]
    [Column("tid")]
    public override int Tid { get; init; }

    [Required]
    [StringLength(50)]
    [Column("title")]
    public string? Title { get; init; }

    [Required]
    [StringLength(120)]
    [Column("description")]
    public string? Description { get; init; }

    [Required]
    [Column("start_time")]
    public DateTime StartTime { get; init; }

    [Required]
    [StringLength(30)]
    [Column("username")]
    public string? Username { get; init; }

    private bool Equals(Task other)
    {
        return base.Equals(other) && Tid == other.Tid && Title == other.Title && Description == other.Description && StartTime.Equals(other.StartTime) && Username == other.Username;
    }

    public override bool Equals(object? obj)
    {
        if (ReferenceEquals(null, obj)) return false;
        if (ReferenceEquals(this, obj)) return true;
        return obj.GetType() == GetType() && Equals((Task)obj);
    }

    public override int GetHashCode()
    {
        return HashCode.Combine(base.GetHashCode(), Tid, Title, Description, StartTime, Username);
    }

    public override string ToString()
    {
        return
            $"{base.ToString()}, {nameof(Tid)}: {Tid}, {nameof(Title)}: {Title}, {nameof(Description)}: {Description}, {nameof(StartTime)}: {StartTime}, {nameof(Username)}: {Username}";
    }
}

File: EmployeeWatch-Java-Project/src/main/java/com/iss/Domain/Task.java
package com.iss.Domain;

import java.time.LocalDateTime;
import java.util.*;

/**
 *
 */
public class Task extends Entity<Integer> {
    private final String description;
    private final LocalDateTime startTime;

    public Task(Integer id, String description, LocalDateTime startTime) {
        super(id);
        this.description = description;
        this.startTime = startTime;
    }

    public String getDescription() {
        return description;
    }

    public LocalDateTime getStartTime() {
        return startTime;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        if (!super.equals(o)) return false;
        Task task = (Task) o;
        return Objects.equals(description, task.description) && Objects.equals(startTime, task.startTime);
    }

    @Override
    public int hashCode() {
        return Objects.hash(super.hashCode(), description, startTime);
    }

    @Override
    public String toString() {
        return "Task{" +
                "description='" + description + '\'' +
                ", startTime=" + startTime +
                "} " + super.toString();
    }
}